### PR TITLE
Dev keyvault microsoft.key vault 2022 02 01 preview

### PR DESCRIPTION
**Pull request by rahulalapati43**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by TimLovellSmith** [comment]
Why is the PR touching so many files, like Microsoft.ApiManagement, Microsoft.AppConfiguration, Microsoft.Cache, if its a keyvault PR?? Can you please clean it up?

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/common.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        },
+        "message": {
+          "type": "string",
+          "description": "User friendly error message. The message is typically localized and may vary with service version."
+        }
+      },
+      "x-ms-external": true,
+      "type": "object"
+    },
+    "SystemData": {
```
Comment (TimLovellSmith) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/common.json:35]: >SystemData [](http://example.com/codeflow?start=5&length=10)

If you change to referencing the systemData definition in common-types/v4 would it be non-breaking? Can you do that instead?

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json [hunk 2]
```diff
@@ -0,0 +1,2064 @@
...
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Private endpoint connection resource.",
+      "x-ms-azure-resource": true,
+      "type": "object"
+    },
+    "PrivateEndpointConnectionListResult": {
```
Comment (TimLovellSmith) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json:1854]: >PrivateEndpointConnectionListResult [](http://example.com/codeflow?start=5&length=35)

If you changed to referencing the private endpoint connection definitions in common-types/v*/privatelinks.json would it be non-breaking? If so, can you refactor to that instead?

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json [hunk 3]
```diff
@@ -0,0 +1,2064 @@
...
+          },
+          "description": "Required DNS zone names of the the private link resource."
+        }
+      },
+      "description": "Properties of a private link resource.",
+      "type": "object"
+    }
+  },
+  "parameters": {
+    "SubscriptionIdParameter": {
```
Comment (TimLovellSmith) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json:2013]: >SubscriptionIdParameter [](http://example.com/codeflow?start=5&length=23)

If you remove this and instead reference the SubscriptionIdParameter in common-types/v4/types.json would it be non-breaking? If so, can you please do that instead?

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json [hunk 4]
```diff
@@ -0,0 +1,2064 @@
...
+      "name": "subscriptionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Subscription credentials which uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call."
+    },
+    "ResourceGroupName": {
```
Comment (TimLovellSmith) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json:2020]: >ResourceGroupName [](http://example.com/codeflow?start=5&length=17)

If you remove this and instead reference the ResourceGroupNameParameter in common-types/v4/types.json would it be non-breaking? If so, can you please do that instead?

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json [hunk 5]
```diff
@@ -0,0 +1,2064 @@
...
+      "x-ms-parameter-location": "method"
+    },
+    "PrivateEndpointConnectionName": {
+      "name": "privateEndpointConnectionName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Name of the private endpoint connection associated with the key vault.",
+      "x-ms-parameter-location": "method"
+    },
```
Comment (TimLovellSmith) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json:2044]: Looks like we should get this added in common-types too if it is not there....

Comment (TimLovellSmith) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json:2044]: It is already there, just in the privatelinks.json:

          {
            "$ref": "../../../../../common-types/resource-management/v3/privatelinks.json#/parameters/PrivateEndpointConnectionName"
          }

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json [hunk 6]
```diff
@@ -0,0 +1,2064 @@
...
+    "ApiVersionParameter": {
```
Comment (TimLovellSmith) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/keyvault.json:2045]: >ApiVersionParameter [](http://example.com/codeflow?start=5&length=19)

Factoring - If you remove this and instead reference the ApiVersionParameter in common-types/v4/types.json would it be non-breaking? If so, can you please do that instead?

**Comment by TimLovellSmith** [comment]
@pilor is added to the review. #Closed

**Comment by TimLovellSmith** [comment]
@pilor This API review includes soft delete APIs, would you might be interested to review it for API consistency?

**Comment by pilor** [comment]
> @pilor This API review includes soft delete APIs, would you be interested to review it for API consistency?

Looks like it was just moved in the spec and not significantly changed.  KeyVault already has an existing soft delete pattern that they should continue to follow and this doesn't appear to be changing that existing pattern.

**Comment by rahulalapati43** [comment]
> Why is the PR touching so many files [mainly whitespace], like Microsoft.ApiManagement, Microsoft.AppConfiguration, Microsoft.Cache, if its a keyvault PR?? Can you please clean it up?



> Why is the PR touching so many files [mainly whitespace], like Microsoft.ApiManagement, Microsoft.AppConfiguration, Microsoft.Cache, if its a keyvault PR?? Can you please clean it up?

@TimLovellSmith All the changes to non KeyVault files, like Microsoft.ApiManagement, Microsoft.AppConfiguration, Microsoft.Cache, EventGrid, etc are prettier fixes when I ran the npm run lint --fix command.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/common.json (hunk 1 above)
Comment (rahulalapati43) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2022-02-01-preview/common.json:35]: @TimLovellSmith Quick background regarding these changes:

- The changes to the KeyVault files were made to fix the S360 issues related to swagger completeness.
- I had created a dev branch out of the main branch, then created another branch from dev branch. Then created a PR from my branch to dev branch: https://github.com/Azure/azure-rest-api-specs/pull/22012.
- I had fixed all the errors reported in the CI pipeline stages including the breaking changes. There was one stage [SDK azure-powershell](https://github.com/Azure/azure-rest-api-specs/pull/22012/checks?check_run_id=10427914861) which was failing, but the Swagger team confirmed that the issue was not related to my changes.
- Could you please help me understand, why I am running into new undetected errors in this CI pipeline, which were undetected in the earlier PR to dev branch?

**Pull request closed without merging**